dev/benchmarks/macrobenchmarks/lib/src/web/bench_mouse_region_grid_hover.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'recorder.dart';
+
+class _NestedMouseRegion extends StatelessWidget {
+  const _NestedMouseRegion({this.nests, this.child});
+
+  final int nests;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    Widget current = child;
+    for (int i = 0; i < nests; i++) {
+      current = MouseRegion(
+        onEnter: (_) {},
+        child: child,
+      );
+    }
+    return current;
+  }
+}
+
+/// Creates a grid of mouse regions, then continuously hover over them.
+///
+/// Measures our ability to hit test mouse regions.
+class BenchMouseRegionGridHover extends WidgetRecorder {
+  BenchMouseRegionGridHover() : super(name: benchmarkName);
+
+  static const String benchmarkName = 'bench_mouse_region_grid_hover';
+
+  final _Tester tester = _Tester();
+
+  // Use a non-trivial border to force Web to switch painter
```

Comment:
Switch painter in what way?

Reply:
Doesn't Web use a simpler painter (CSS?) for simple widgets, and canvas for more advanced ones? According to my test the performance only regresses when the box has a partial border.